### PR TITLE
Add handling for the query path

### PR DESCRIPTION
**Pull request by jackzampolin**

### CODE REVIEW

**Comment by joelegasse** [comment]
This needs to be discussed, designed, and would need a re-architecture of the current codebase. Please open an issue for discussion, first.

**Comment by nhaugo** [comment]
I have created the issue at #51

File: relay/http.go [hunk 1]
```diff
@@ -113,13 +114,15 @@ func (h *HTTP) Stop() error {
...
+	reqPath := r.URL.Path
+
+	if reqPath == "/ping" && (r.Method == "GET" || r.Method == "HEAD") {
+		w.Header().Add("X-InfluxDB-Version", "relay")
+		w.WriteHeader(http.StatusNoContent)
+		return
 	}
 
-	if r.URL.Path != "/write" {
+	if reqPath != "/write" && reqPath != "/query" {
```
Comment (joelegasse) [inline relay/http.go]: This should now be handled with a request router, and has gone grown far beyond a simple path check.

File: relay/http.go [hunk 2]
```diff
@@ -136,6 +139,23 @@ func (h *HTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	queryParams := r.URL.Query()
 
+	// Don't pass through non create queries
+	if reqPath == "/query" {
+		// First check if there are multiple queries being passed
+		queries := strings.Split(queryParams["q"][0], ";")
+		if len(queries) > 1 {
+			// If there are drop any queries other than the first
+			queryParams["q"][0] = queries[0]
```
Comment (joelegasse) [inline relay/http.go]: silently dropping the query string, then returning a success is definitely the wrong thing to do. multiple statements should either be rejected, or processed individually.

File: relay/http.go [hunk 3]
```diff
@@ -209,7 +229,11 @@ func (h *HTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		b := b
 		go func() {
 			defer wg.Done()
-			resp, err := b.post(outBytes, query, authHeader)
+			q := false
+			if reqPath == "/query" {
```
Comment (joelegasse) [inline relay/http.go]: checking this kind of things multiple times throughout the code further reinforces that a re-structure is needed, and this PR is not a suitable implementation

File: relay/http.go [hunk 4]
```diff
@@ -312,8 +336,17 @@ func newSimplePoster(location string, timeout time.Duration, skipTLSVerification
 	}
 }
 
-func (b *simplePoster) post(buf []byte, query string, auth string) (*responseData, error) {
-	req, err := http.NewRequest("POST", b.location, bytes.NewReader(buf))
+func (b *simplePoster) post(buf []byte, query string, auth string, q bool) (*responseData, error) {
+	loc := b.location
+	if q {
+		location, err := url.Parse(b.location)
```
Comment (joelegasse) [inline relay/http.go]: re-parsing a pre-computed value, then modifying the result to get a different string is definitely a poor design that would need to be removed

**Pull request merged**